pkg/manager/member/executor_manager.go

```diff
@@ -725,7 +718,158 @@ func (m *executorMemberManager) getContainerByName(sts *appsv1.StatefulSet, cont
+		// get endpoints info
+		eps := &corev1.Endpoints{}
+		epErr := m.cli.Get(context.TODO(), types.NamespacedName{
+			Namespace: ns,
+			Name:      controller.TiflowMasterMemberName(tcName),
+		}, eps)
+		if epErr != nil {
+			return nil, fmt.Errorf("syncTiflowClusterStatus: failed to get endpoints %s for cluster %s/%s, err: %s, epErr %s",
+				controller.TiflowMasterMemberName(tcName), ns, tcName, err, epErr)
+		}
+		if eps != nil && len(eps.Subsets) == 0 {
+			return nil, fmt.Errorf("%s, service %s/%s has no endpoints",
+				err, ns, controller.TiflowMasterMemberName(tcName))
+		}
+		return nil, err
+	}
+
+	return syncExecutorMembers(executorsInfo)
+}
+
+func (m *executorMemberManager) syncExecutorsStatusWithHetero(tc *v1alpha1.TiflowCluster) (map[string]v1alpha1.ExecutorMember, error) {
+	ns := tc.Spec.Cluster.Namespace
+	tcName := tc.Spec.Cluster.Name
+
+	hetero := tiflowapi.Heterogeneous{}
+	if tc.Heterogeneous() && tc.Spec.Master == nil {
+		hetero.Hetero = true
+		hetero.Namespace = ns
+		hetero.Name = tcName
+	}
+
+	tiflowClient := tiflowapi.GetMasterClient(m.cli, ns, tcName, "", tc.IsClusterTLSEnabled(), hetero)
+
+	// get executors info from master
+	executorsInfo, err := tiflowClient.GetExecutors()
+	if err != nil {
+		tc.Status.Executor.Synced = false
+		// get endpoints info
+		eps := &corev1.Endpoints{}
+		epErr := m.cli.Get(context.TODO(), types.NamespacedName{
+			Namespace: ns,
+			Name:      controller.TiflowMasterMemberName(tcName),
+		}, eps)
+		if epErr != nil {
+			return nil, fmt.Errorf("syncTiflowClusterStatus: failed to get endpoints %s for cluster %s/%s, err: %s, epErr %s",
+				controller.TiflowMasterMemberName(tcName), ns, tcName, err, epErr)
+		}
+		if eps != nil && len(eps.Subsets) == 0 {
+			return nil, fmt.Errorf("%s, service %s/%s has no endpoints",
+				err, ns, controller.TiflowMasterMemberName(tcName))
+		}
+		return nil, err
+	}
+	return syncExecutorMembers(executorsInfo)
+}
+
+func syncExecutorMembers(executors tiflowapi.ExecutorsInfo) (map[string]v1alpha1.ExecutorMember, error) {
+	// todo: WIP, get information about the FailureMembers and FailoverUID through the MasterClient
+	// sync executors info
+	members := make(map[string]v1alpha1.ExecutorMember)
+	for _, e := range executors.Executors {
+		c, err := handleCapability(e.Capability)
+		if err != nil {
+			return nil, err
+		}
+
+		memberName, err := formatExecutorName(e.ID)
+		if err != nil {
+			return nil, err
+		}
+
+		members[memberName] = v1alpha1.ExecutorMember{
+			Id:         e.ID,
+			Name:       e.Name,
+			Addr:       e.Address,
+			Capability: c,
+		}
+	}
+
+	return members, nil
+}
+
+func handleCapability(o string) (int64, error) {
+	var i interface{}
+	d := json.NewDecoder(strings.NewReader(o))
+	d.UseNumber()
+
+	if err := d.Decode(&i); err != nil {
+		return -1, err
+	}
+
+	n := i.(json.Number)
+	res, err := n.Int64()
+	if err != nil {
+		return -1, err
+	}
+
+	return res, nil
+}
+
+func formatExecutorName(name string) (string, error) {
```

Comment:
Why don't use name directly? This information will be shown to users.

Reply:
formatName is the key information in the map structure, and the specific information exists in the value, which is visible to the user. In addition, master currently has no name information, so you need to add it yourself

Reply:
Then I think just using `podName` or `podName.namespace` is enough.
`fmt.Sprintf("%s/Executor: %s", nameSlice[2], nameSlice[0])` is a bit complicated.